synapse/storage/databases/main/stream.py

```diff
@@ -795,6 +797,32 @@ async def get_current_room_stream_token_for_room_id(
         )
         return RoomStreamToken(topo, stream_ordering)
 
+    @overload
+    def get_stream_id_for_event_txn(
+        self,
+        txn: LoggingTransaction,
+        event_id: str,
+        allow_none: Literal[True],
+    ) -> int:
+        ...
+
+    @overload
+    def get_stream_id_for_event_txn(
+        self,
+        txn: LoggingTransaction,
+        event_id: str,
+    ) -> int:
+        ...
```

Comment:
Can we get away with one fewer overload if we include the default value for `allow_none`?
```suggestion
    @overload
    def get_stream_id_for_event_txn(
        self,
        txn: LoggingTransaction,
        event_id: str,
        allow_none: Literal[True] = True,
    ) -> int:
        ...
```